core/src/main/java/org/apache/calcite/avatica/util/ArrayImpl.java

```diff
@@ -220,6 +220,17 @@ Object getArrayData(Object o, AbstractCursor.ArrayAccessor componentAccessor)
     // nothing to do
   }
 
+  public static boolean equalContents(Array left, Array right) throws SQLException {
+    ResultSet leftResultSet = left.getResultSet();
+    ResultSet rightResultSet = right.getResultSet();
+    while (leftResultSet.next() && rightResultSet.next()) {
+      if (!leftResultSet.getObject(1).equals(rightResultSet.getObject(1))) {
```

Comment:
Please switch this over to be `Objects.equals(leftResultSet.getObject(1), rightResultSet.getObject(1))`. Just in case either of these are ever null :)

Reply:
Done

Reply:
> Just in case either of these are ever null

And what I really meant to say was "in case `leftResultSet.getObject(1)` is ever null" 🙃